### DISCUSSION TITLE
`doi:` is overriding `url:` in the html citation

### DISCUSSION THREAD

**Discussion by JoFrhwld** [quarto-dev/quarto-cli#2739] [Q&A] [answered]
I'm trying to put together a website where some of the pages will have DOIs minted for them, and would like to include both the url and the doi in the recommended citation. The problem I'm running into is that if a DOI is specified for a page, only the link to doi.org is included in the html citation. 

I have an example project here: https://github.com/JoFrhwld/doi_url

In `index.qmd` I define a doi [here](https://github.com/JoFrhwld/doi_url/blob/8d75bcf1c4fe4e6585e7142cf705e91dd0014d8e/index.qmd#L5)

The bibtex comes out fine:

```
@online{doe2022,
  author = {Jane Doe},
  editor = {},
  title = {Doi\_url},
  date = {2022-10-05},
  url = {http://www.example.com//},
  doi = {10.23915/reprodocs.00010},
  langid = {en}
}
```

But the html citation comes out as:

> Jane Doe. 2022. “Doi_url.” October 5, 2022. https://doi.org/10.23915/reprodocs.00010.

In `url_page.qmd`, I don't define a doi, and again the bibtex is fine, and now I have the page url in the html citation:

> Jane Doe. 2022. “Url_page.” October 5, 2022. http://www.example.com//url_page.html.

I'm not sure where to start digging into to customize or adjust the output of the citation so that both the doi and the url are shown. Any help would be appreciated!

**Comment by dragonstyle** [comment, marked as answer]
The formatted citation at the bottom of the page is driven by the CSL used for the page (e.g. we will use the page CSL to create the formatted citation). 

```markdown
---
title: Test
csl: american-journal-of-health-behavior.csl
...
---
```

On one hand, this is great, because there is a large library of CSL styles available (<https://www.zotero.org/styles>), but it also may make customizing the appearance of the reference a little more challenging as you either will need to create a CSL style that matches your goal, or find one that you think works.

**Reply by JoFrhwld** [reply]
Ah! This is great news actually! Is specifying a custom csl as easy as adding it to the project and pointing to it in the project _quarto.yml?

**Reply by dragonstyle** [reply]
Yep, it should be that easy!

**Reply by JoFrhwld** [reply]
Awesome thanks!

**Reply by cscheid** [reply]
@dragonstyle do we have a minimal example of this to add to quarto-examples?

**Reply by rgaiacs** [reply]
@JoFrhwld be careful when customising CSL files as they can define "complex" rules, for example: insert `x` if BibTeX entry has field `x`, insert `y` if BibTeX entry has field `y`, ...

**Reply by dragonstyle** [reply]
@cscheid I added an example but am not sure how to publish it / not sure I have credentials to do so. Do you know how?

**Reply by cscheid** [reply]
I don't; I think @cderv has them, but he's out for the immediate future. The repo is good enough for now, thank you!!

**Reply by cderv** [reply]
Passing by around here, and it was 2min to publish. So it is now live at https://examples.quarto.pub/csl-example/
